localflavor/ch/forms.py

```diff
@@ -135,3 +136,31 @@ def clean(self, value):
             raise ValidationError(self.error_messages['invalid'])
 
         return '%s%s%s' % (idnumber, pos9, checksum)
+
+
+class CHSocialSecurityNumberField(forms.RegexField):
+    """
+    A Swiss Social Security number (also known as the new AHV Number).
+
+    Checks the following rules to determine whether the number is valid:
+
+        * Conforms to the 756.XXXX.XXXX.XX
+        * Included checksums match calculated checksums
+
+    See:
+    http://de.wikipedia.org/wiki/Sozialversicherungsnummer#Versichertennummer
+    """
+    default_error_messages = {
+        'invalid': _('Enter a valid Swiss Social Security number in 756.XXXX.XXXX.XX format.'),
+    }
+
+    def __init__(self, max_length=None, min_length=None, *args, **kwargs):
+        super(CHSocialSecurityNumberField, self).__init__(ssn_re, max_length, min_length, *args, **kwargs)
```

Comment:
This has been my initial approach, but then you get duplicated error messages as all the validators are run together (maybe I missed something...). In this case the user experience is not that nice, since you would get a warning for instance about the failed EAN checksum validation, I'd prefer to just have the error message from `CHSocialSecurityNumberField`. Any suggestions?